Source/Base/Base/Math/Interpolation.h

```diff
@@ -0,0 +1,403 @@
+                    {
+                        t -= 1.5f / d1;
+                        return n1 * t * t + 0.75f;
+                    }
+                    else if (t < 2.5f / d1)
+                    {
+                        t -= 2.25f / d1;
+                        return n1 * t * t + 0.9375f;
+                    }
+                    else
+                    {
+                        t -= 2.625f / d1;
+                        return n1 * t * t + 0.984375f;
+                    }
+                }
+
+                static f32 Polynomial(f32 t, i32 degree)
+                {
+                    ClampAnim(t);
+                    ClampPolynomialDegree(degree);
+                    return 1.0f - pow(1.0f - t, static_cast<f32>(degree));
+                }
+
+                static f32 Quadratic(f32 t) { return Polynomial(t, 2); }
+                static f32 Cubic(f32 t)     { return Polynomial(t, 3); }
+                static f32 Quartic(f32 t)   { return Polynomial(t, 4); }
+                static f32 Quintic(f32 t)   { return Polynomial(t, 5); }
+                static f32 Sextic(f32 t)    { return Polynomial(t, 6); }
+                static f32 Septic(f32 t)    { return Polynomial(t, 7); }
+                static f32 Octic(f32 t)     { return Polynomial(t, 8); }
+            };
+
+            struct In
+            {
+                static f32 Sine(f32 t)
+                {
+                    ClampAnim(t);
+                    return 1.0f - Math::Cos((t * Math::PI) / 2.0f);
+                }
+
+                static f32 Circ(f32 t)
+                {
+                    ClampAnim(t);
+                    return 1.0f - Math::Sqrt(1.0f - pow(t, 2.0f));
+                }
+
+                static f32 Elastic(f32 t)
+                {
+                    ClampAnim(t);
+                    if (t == 0.0f || t == 1.0f)
+                        return t;
+
+                    f32 c4 = (2.0f * Math::PI) / 3.0f;
+                    return -pow(2.0f, 10.0f * t - 10.0f) * Math::Sin((t * 10.0f - 10.75f) * c4);
+                }
+
+                static f32 Expo(f32 t)
+                {
+                    ClampAnim(t);
+                    if (t == 0.0f)
+                        return t;
+
+                    return pow(2.0f, 10.0f * t - 10.0f);
+                }
+
+                static f32 Back(f32 t)
+                {
+                    ClampAnim(t);
+                    f32 c1 = 1.70158f;
+                    f32 c3 = c1 + 1.0f;
+
+                    return c3 * t * t * t - c1 * t * t;
+                }
+
+                static f32 Bounce(f32 t)
+                {
+                    ClampAnim(t);
+                    return 1.0f - Ease::Out::Bounce(1.0f - t);
+                }
+
+                static f32 Polynomial(f32 t, i32 degree)
+                {
+                    ClampAnim(t);
+                    ClampPolynomialDegree(degree);
+                    return pow(t, static_cast<f32>(degree));
```

Comment:
I'd prefer if we made the degree as float a variable instead of directly doing the cast when the function paramter is passed.